packages/example-codehub/src/codehub-application.ts

```diff
@@ -3,23 +3,32 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Application, CoreBindings} from '@loopback/core';
+import {Application, ApplicationConfig} from '@loopback/core';
+import {RestServer, RestBindings} from '@loopback/rest';
 import {UserController, HealthController} from './controllers';
 
 export class CodeHubApplication extends Application {
-  constructor() {
-    super();
+  constructor(options?: ApplicationConfig) {
+    super(options);
 
     const app = this;
-
-    app.api({
-      swagger: '2.0',
-      info: {
-        title: 'CodeHub',
-        version: require('../package.json').version,
-      },
-      basePath: '/',
-      paths: {},
+    app.server(RestServer);
+    // We can't use the async version in the ctor!
+    const server = app.getServer(RestServer);
+
+    // tslint:disable-next-line:no-floating-promises
+    server.then(s => {
+      s.api({
+        swagger: '2.0',
+        info: {
+          title: 'CodeHub',
+          version: require('../package.json').version,
+        },
+        basePath: '/',
+        paths: {},
+      });
+
+      s.bind(RestBindings.PORT).to(3000);
```

Comment:
Oh, I see, you are binding the Rest PORT on the server object. IMO, it should be possible to configure this binding on the application object, and `RestBindings.PORT` should use a namespaced key to avoid conflicts.

Ideally, it should be also possible to retrieve the updated port from the `app` object too (for consistency and ease of use). However, I think that would require us to improve Context/Binding API to allow in-place edit of a bound value in the context where the value was bound, which would only inflate the scope of this pull request, so let's leave it out (but please create a follow-up GH issue if you agree this should be done).